mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/builtin/impl/DefaultJavaNetURLBridge.java

```diff
@@ -30,9 +32,9 @@ protected String toString(URL value) {
 	@Override
 	protected URL fromString(String value) {
```

Comment:
If you update `fromString`, maybe you should also update `toString` (using `toURI().toString()` instead of `toExternalForm()`). Mainly because of how URL/URI handle escapes differently.

Perhaps that's not necessary, though. In any case, could you add a few values to `JavaNetURLPropertyTypeDescriptor` to test URL escaping? I would also include an URL with an URL escape that has been created with `new URL`, just in case, because users could do that.